src/promptflow-tools/promptflow/tools/open_source_llm.py

```diff
@@ -76,6 +77,95 @@ def format_generic_response_payload(output: bytes, response_key: str) -> str:
         raise OpenSourceLLMUserError(message=message)
 
 
+def get_model_type(deployment_model: str) -> str:
+    m = re.match(r'azureml://registries/[^/]+/models/([^/]+)/versions/', deployment_model)
+    if m is None:
+        raise ValueError(f"Unexpected model format: {deployment_model}")
+    model = m[1].lower()
+    if model.startswith(ModelFamily.LLAMA.lower()):
+        return ModelFamily.LLAMA
+    elif model.startswith(ModelFamily.FALCON.lower()):
+        return ModelFamily.FALCON
+    elif model.startswith(ModelFamily.DOLLY.lower()):
+        return ModelFamily.DOLLY
+    elif model.startswith("gpt2"):
+        return ModelFamily.GPT2
+    else:
+        raise ValueError(f"Unexpected model type: {model} derived from deployed model: {deployment_model}")
+
+
+def get_deployment_from_endpoint(endpoint_name: str, deployment_name: str = None) -> Tuple[str, str, str]:
+    from azure.identity import DefaultAzureCredential
+    credential = DefaultAzureCredential(exclude_interactive_browser_credential=False)
+
+    try:
+        from azure.ai.ml import MLClient
+        ml_client = MLClient(
+            credential=credential,
+            subscription_id=os.getenv("AZUREML_ARM_SUBSCRIPTION"),
```

Comment:
Who will set the os env?

Reply:
They would have to - I'm not aware of any other way to get the workspace details, do you know of any @chjinche?